pkg/controller/kubelet-config/helpers.go

```diff
@@ -184,11 +179,9 @@ func updateMachineConfigwithCgroup(node *osev1.Node, mc *mcfgv1.MachineConfig) e
 	case osev1.CgroupModeV1:
 		kernelArgsToAdd = append(kernelArgsToAdd, kernelArgsv1...)
 		kernelArgsToRemove = append(kernelArgsToRemove, kernelArgsv2...)
-	case osev1.CgroupModeV2:
+	case osev1.CgroupModeV2, osev1.CgroupModeEmpty:
```

Comment:
@sairameshv Can you evaluate if this change plays well with `WorkerLatencyProfiles`?

Reply:
As of today, 97-master/worker-generated-kubelet mc is present from Day-0
Hence, this is working as expected from the workerlatency update perspective.
However, if we don't want the above MC present by default in the cluster from Day-0, this change actually doesn't return if the nodes.config object's .spec is empty and hence creates the unwanted 97-master/worker-generated-kubelet mcs

Reply:
@yuqi-zhang, Is my understanding correct?

Reply:
An empty spec needs to still set kernel arguments to turn on PSI, because an empty spec is really "CgroupModeEmpty" which needs to default to cgroupv2.

The way this is written will skip turning on PSI or setting any kernel arguments.

Reply:
@sairameshv I think your understanding is correct. I suppose in a sense there is no harm to having it being explicitly set if it works around other issues, but today if no kargs are generated we just use the RHCOS9 default of v2 on the node.

>An empty spec needs to still set kernel arguments to turn on PSI, because an empty spec is really "CgroupModeEmpty" which needs to default to cgroupv2.

>The way this is written will skip turning on PSI or setting any kernel arguments.

Could I ask how what PSI is and how it is being used?

Reply:
The default MCs after this patch are:

```
NAME                                               GENERATEDBYCONTROLLER                      IGNITIONVERSION   AGE
00-master                                          fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             6m11s
00-worker                                          fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             6m11s
01-master-container-runtime                        fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             6m11s
01-master-kubelet                                  fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             6m11s
01-worker-container-runtime                        fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             6m11s
01-worker-kubelet                                  fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             6m11s
42-autologin-master                                                                           2.2.0             8m40s
42-autologin-worker                                                                           2.2.0             8m40s
97-master-generated-kubelet                        fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             6m11s
97-worker-generated-kubelet                        fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             6m11s
98-master-generated-kubelet                        fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             6m11s
98-worker-generated-kubelet                        fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             6m9s
99-master-generated-registries                     fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             6m11s
99-master-ssh                                                                                 3.2.0             8m41s
99-worker-generated-registries                     fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             6m11s
99-worker-ssh                                                                                 3.2.0             8m41s
rendered-master-2add363fefc6418f96694487acb36441   fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             6m7s
rendered-worker-aed871084d805eba819a369887d5bc53   fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             6m7s
```

Reply:
The default MCs after this patch (cgroupsv1, Node Object at Day 0):

```
NAME                                               GENERATEDBYCONTROLLER                      IGNITIONVERSION   AGE
00-master                                          fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             10m
00-worker                                          fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             10m
01-master-container-runtime                        fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             10m
01-master-kubelet                                  fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             10m
01-worker-container-runtime                        fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             10m
01-worker-kubelet                                  fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             10m
42-autologin-master                                                                           2.2.0             12m
42-autologin-worker                                                                           2.2.0             12m
97-master-generated-kubelet                        fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             10m
97-worker-generated-kubelet                        fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             10m
98-master-generated-kubelet                        fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             10m
98-worker-generated-kubelet                        fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             10m
99-master-generated-registries                     fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             10m
99-master-ssh                                                                                 3.2.0             12m
99-worker-generated-registries                     fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             10m
99-worker-ssh                                                                                 3.2.0             12m
rendered-master-289c7230546777a88ede292224d02737   fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             10m
rendered-worker-10bf0407840448c817a78e436f7ae11a   fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             10m
```

Reply:
@yuqi-zhang PSI is a Linux Kernel feature to propagate information on which cgroups/processes have 'contention' with memory, cpu, or IO. More info [here](https://docs.kernel.org/accounting/psi.html). It needs to be explicitly turned on with psi=1 to be enabled.

Reply:
The default MCs after this patch+NTO patch (NTO object at day 0), defaults to cgroupv1 post-pivot.

```
NAME                                               GENERATEDBYCONTROLLER                      IGNITIONVERSION   AGE
00-master                                          fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             8m54s
00-worker                                          fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             8m54s
01-master-container-runtime                        fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             8m54s
01-master-kubelet                                  fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             8m54s
01-worker-container-runtime                        fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             8m54s
01-worker-kubelet                                  fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             8m54s
42-autologin-master                                                                           2.2.0             11m
42-autologin-worker                                                                           2.2.0             11m
50-nto-worker                                                                                 3.2.0             3m14s
50-performance-clusterbotpp                                                                   3.2.0             3m23s
97-master-generated-kubelet                        fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             8m53s
97-worker-generated-kubelet                        fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             8m53s
98-master-generated-kubelet                        fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             8m54s
98-worker-generated-kubelet                        fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             8m54s
99-master-generated-registries                     fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             8m54s
99-master-ssh                                                                                 3.2.0             11m
99-worker-generated-kubelet                        fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             3m22s
99-worker-generated-registries                     fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             8m54s
99-worker-ssh                                                                                 3.2.0             11m
rendered-master-9a207760103bdda3699bad3edc2ad5d4   fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             8m51s
rendered-worker-127f4a7e7f9fd17b65adbff282848f22   fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             3m18s
rendered-worker-aaa98bd21ed456fde0f28f7ce5a8a26a   fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             8m51s
rendered-worker-c17e218c7813ca922a4efaca6b586e92   fb7c391a3b47cf566998881ce9ceaaaed4d820a9   3.4.0             3m13s
```

Reply:
Ok, essentially then we're saying for all clusters regardless of setting node configs or not, we will explicitly set your kargs via machineconfigs for cgroups v2 (or you can explicitly set to v1 yourself).

If we're ok with that then I think it should be fine.